js/5etools-main.js

```diff
@@ -1537,6 +1582,17 @@ const betteR205etoolsMain = function () {
 					attrs.add(`repeating_tool_${tRowID}_toolattr_base`, "?{Attribute?|Strength,@{strength_mod}|Dexterity,@{dexterity_mod}|Constitution,@{constitution_mod}|Intelligence,@{intelligence_mod}|Wisdom,@{wisdom_mod}|Charisma,@{charisma_mod}}");
 					attrs.add(`repeating_tool_${tRowID}_toolbonus_display`, "?");
 				});
+
+				// Add flavor traits
+				if (traits && traits.length == 4) {
```

Comment:
as mentioned above, this could use object destructuring to become e.g.

```
const {personalityTraits, ideals, bonds, flaws} = traits || {};
if (personalityTraits) { ... add traits ... }
if (ideals) { ... add ideal ... }
...
```

which would also allow the user to enter partial information, if that was something you ultimately wanted to support